extensions/wikia/SEOTweaks/SEOTweaksHooksHelper.class.php

```diff
@@ -284,6 +284,34 @@ static public function onArticleViewHeader( Article $article, bool &$outputDone,
 		return true;
 	}
 
+	/**
+	 * When #REDIRECT is used make a proper 301 redirect
+	 *
+	 * @param Title $title
+	 * @param $unused
+	 * @param OutputPage $output
+	 * @param User $user
+	 * @param WebRequest $request
+	 * @param MediaWiki $mediawiki
+	 * @return bool
+	 */
+	static public function onBeforeInitialize(
+		Title $title, $unused, OutputPage $output,
+		User $user, WebRequest $request, MediaWiki $mediawiki
+	): bool {
+		if ( !$user->isAnon() || !$title->isRedirect() ) {
+			return true;
+		}
+
+		$queryParams = $request->getQueryValues();
+		unset( $queryParams['title'] );
+		$targetUrl = $output->getWikiPage()->getRedirectTarget()->getFullURL( $queryParams );
+
+		$output->redirect( $targetUrl, HTTP_REDIRECT_PERM );
```

Comment:
Please use the third param of the  `redirect` method - the ` $redirectedBy`. This identifies the code that triggered a redirect.